### PR TITLE
Add github libraries plugin

### LINKED ISSUE #14: Add commands to request libraries by languages
A nifty feature since the libraries topic is extremely popular, is add commands like `!libs php` or `!libs js` that would fetch a list of libraries by language. This might be helpful when someone comes into the channel and tries to get help for, for example Java, and would find a dead end because the java devs happen to be offline.

### PR DESCRIPTION
**Pull request by stelar7**
Fixes #14

### CODE REVIEW

File: src/GithubLibraries.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        if (!command.startsWith("!")) {
+            return;
+        }
+
+        if (this.settings.githubLibraries.aliases.some(x => x === command.substr(1))) {
+
+            const response = await fetch(this.settings.githubLibraries.baseURL + language);
+            const data = await response.json();
+
+            if (this.isValidResponse(data)) {
```
Comment (molenzwiebel) [inline src/GithubLibraries.ts]: `!` missing?

File: src/GithubLibraries.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                message.reply(this.settings.githubLibraries.noLanguage + language);
+                return;
+            }
+
+            var printMe = `List of languages for ${language}:\n`;
```
Comment (molenzwiebel) [inline src/GithubLibraries.ts]: `var`

Comment (molenzwiebel) [reply src/GithubLibraries.ts]: Also, should this be "List of libraries", not "List of languages"?

File: src/GithubLibraries.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            data.map(await this.readJsonData).forEach(x => printMe += x);
```
Comment (molenzwiebel) [inline src/GithubLibraries.ts]: Is this valid syntax even? A better/clearer way would probably be:
```ts
for (const lib of data) {
    printMe += await this.readJsonData(lib);
}
```

File: src/GithubLibraries.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    async readJsonData(json: GithubAPIStruct): Promise<string> {
+        const response = await fetch(json.download_url);
+        const data = await response.json();
+
+        if (!this.isValidLibrary(data)) {
+            return "";
+        }
+
+        const lib = data as APILibraryStruct;
```
Comment (molenzwiebel) [inline src/GithubLibraries.ts]: The type guard in `isValidLibrary` should make this cast/variable not needed.

File: src/app.ts [hunk 5]
```diff
@@ -31,6 +32,7 @@ const forum = new ForumReader(bot.client, sharedSettings, personalSettings, "dat
 //const honeypot = new Honeypot(bot.client, sharedSettings, personalSettings);
 const autoReact = new AutoReact(bot.client, sharedSettings, "data/thinking_data.json", "data/ignored_react_data.json");
 const officeHours = new OfficeHours(bot.client, sharedSettings, "data/office_hours_data.json");
+const githubLibraroes = new GithubLibraries(bot.client, sharedSettings);
```
Comment (molenzwiebel) [inline src/app.ts]: Libraroes

**Comment by molenzwiebel** [comment]
Maybe use markdown linking for the repo? Discord actually support `[this](https://format.com)`

**Comment by Querijn** [comment]
Gonna work on it. Implementing molen's requests, some things I'd like to see and testing.

**Comment by Querijn** [comment]
Done my changes.

**Pull request closed without merging**